virttest/test_setup.py

```diff
@@ -1887,19 +1888,28 @@ def __init__(self, params):
         if distro_obj.name.lower().strip() == 'ubuntu':
             self.polkit_name = "polkitd"
         self.polkitd = service.Factory.create_service(self.polkit_name)
-
-        if self.params.get("action_id"):
-            self.action_id = self.params.get("action_id").split()
-        else:
-            self.action_id = []
         self.user = self.params.get("unprivileged_user")
-        if self.params.get("action_lookup"):
-            # The action_lookup string should be separated by space and
-            # each separated string should have ':' which represent key:value
-            # for later use.
-            self.attr = self.params.get("action_lookup").split()
+        # After libvirt-5.6.0, additional rules are needed.
+        # So case provider needs to provide action_matrix including
+        # all rules.
+        self.action_list = []
+        if (libvirt_version.version_compare(5, 6, 0) and
+                self.params.get("action_matrix")):
+            self.action_list = eval(self.params.get("action_matrix"))
         else:
```

Comment:
I assume we don't if ...else based on libvirt_version.version_compare(5, 6, 0). Just move self.action_list = [] to end of this block, then update the list if libvirt_version.version_compare(5, 6, 0)

Reply:
@chunfuwen , I think your way is also ok. But I have one concern.
Cases will provide action_matrix (for libvirt 5.6.0 and later) and action_id/action_lookup (for libvirt 5.6.0 before). But when libvirt 5.6.0 and later is used, there is no need to handle action_id/action_lookup codes. So I prefer to my current codes. What do you think?

Reply:
I am misled by the comment "After libvirt-5.6.0, additional rules are needed." I assume you need add action_matrix besides previous existing action_id/action_lookup . So If the situation is matched your previous comment, I am fine with your previous solution

Reply:
@chunfuwen ,ok,  I should update the comments. Sorry to make you misunderstand.